client/llb/exec.go

```diff
@@ -719,7 +719,16 @@ func SecretID(id string) SecretOption {
 // SecretAsEnv defines if the secret should be added as an environment variable
 func SecretAsEnv(v bool) SecretOption {
 	return secretOptionFunc(func(si *SecretInfo) {
-		si.IsEnv = v
+		envDefault := ""
+		si.Env = &envDefault
+	})
+}
+
+// SecretAsEnvName defines if the secret should be added as an environment variable
```

Comment:
I kept the original functional option (`SecretAsEnv`) intact to keep backwards compatibility as I think it might already be used.